app/forms/date_form.py

```diff
@@ -16,191 +16,125 @@
     DateCheck,
     OptionalForm,
     DateRequired,
-    MonthYearCheck,
-    YearCheck,
     SingleDatePeriodCheck,
 )
 from app.utilities.schema import load_schema_from_metadata
 
 logger = logging.getLogger(__name__)
 
 
-class DateField(FormField):
-    def __init__(self, answer_store, metadata, answer, error_messages, **kwargs):
-        form_class = get_date_form(answer_store, metadata, answer, error_messages)
-        super(DateField, self).__init__(form_class, **kwargs)
-
-    def process(self, formdata, data=None):
-        if data is not None:
-            substrings = data.split('-')
-            data = {'year': substrings[0], 'month': substrings[1], 'day': substrings[2]}
-
-        super().process(formdata, data)
-
+class DateFormTypes(Enum):
+    YearMonthDay = {'date_format': 'yyyy-mm-dd'}
+    YearMonth = {'date_format': 'yyyy-mm'}
+    Year = {'date_format': 'yyyy'}
 
-class MonthYearField(FormField):
-    def __init__(self, answer_store, metadata, answer, error_messages, **kwargs):
-        form_class = get_month_year_form(answer_store, metadata, answer, error_messages)
-        super(MonthYearField, self).__init__(form_class, **kwargs)
 
-    def process(self, formdata, data=None):
-        if data is not None:
-            substrings = data.split('-')
-            data = {'year': substrings[0], 'month': substrings[1]}
-
-        super().process(formdata, data)
-
-
-class YearField(FormField):
-    def __init__(self, answer_store, metadata, answer, error_messages, **kwargs):
-        form_class = get_year_form(
-            answer_store,
-            metadata,
-            answer,
-            error_messages,
-            kwargs.get('description'),
-            kwargs.get('label'),
+class DateField(FormField):
+    def __init__(
+        self,
+        date_form_type: DateFormTypes,
+        answer_store,
+        metadata,
+        answer,
+        error_messages,
+        **kwargs,
+    ):
+        form_class = get_form(
+            date_form_type, answer, answer_store, metadata, error_messages
         )
-        super(YearField, self).__init__(form_class, **kwargs)
+        super(DateField, self).__init__(form_class, **kwargs)
```

Comment:
You should be able to use `super().__init__(form_class, **kwargs)`. Using the name of the parent class is not needed in python 3.